gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/infra/domain_service/analytics_engine/permissions/ApiAnalyticsQueryFilterDecoratorImplTest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.infra.domain_service.analytics_engine.permissions;
+
+import static io.gravitee.apim.core.analytics_engine.model.FilterSpec.Name.API;
+import static io.gravitee.apim.core.analytics_engine.model.FilterSpec.Name.APPLICATION;
+import static io.gravitee.apim.core.analytics_engine.model.FilterSpec.Operator.EQ;
+import static io.gravitee.apim.core.analytics_engine.model.FilterSpec.Operator.IN;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.gravitee.apim.core.analytics_engine.model.Filter;
+import java.util.*;
+import org.junit.jupiter.api.DisplayNameGeneration;
+import org.junit.jupiter.api.DisplayNameGenerator;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+/**
+ * @author GraviteeSource Team
+ */
+@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
+class ApiAnalyticsQueryFilterDecoratorImplTest {
+
+    @Nested
+    class GetIdsFromFiltersTest {
+
+        @Test
+        void should_get_nothing_when_list_is_empty() {
+            List<Filter> emptyFilters = List.of();
+
+            var ids = ApiAnalyticsQueryFilterDecoratorImpl.getIdsFromFilters(emptyFilters, API);
+
+            assertThat(ids).isEmpty();
+        }
+
+        @Test
+        void should_get_ids_from_EQ_filter() {
+            var apiId = UUID.randomUUID().toString();
+
+            List<Filter> filters = List.of(new Filter(API, EQ, apiId), new Filter(APPLICATION, EQ, UUID.randomUUID().toString()));
+
+            var ids = ApiAnalyticsQueryFilterDecoratorImpl.getIdsFromFilters(filters, API);
+
+            assertThat(ids).containsExactly(apiId);
+        }
+
+        @Test
+        void should_get_ids_from_IN_filter() {
+            var apiId1 = UUID.randomUUID().toString();
+            var apiId2 = UUID.randomUUID().toString();
```

Comment:
I would go with declaring those UUID as strings we can think about when the test fails with a message.